x-pack/plugins/canvas/canvas_plugin_src/functions/common/joinRows.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ExpressionFunction } from '../../../../../../src/legacy/core_plugins/interpreter/types';
+import { Datatable } from '../types';
+import { getFunctionErrors } from '../../strings/functions';
+
+interface Arguments {
+  column: string;
+  separator: string;
+  quote: string;
+}
+
+const escapeString = (data: string, quotechar: string): string => {
+  if (quotechar === undefined || quotechar === '') {
+    return data;
+  } else {
+    const regex = new RegExp(quotechar, 'g');
+    return data.replace(/\\/g, `\\\\`).replace(regex, `\\${quotechar}`);
```

Comment:
The reason I needed to escape it is because if there's e.g. a host.name value that has `'` in it (e.g. `joe's computer`, it will exit the quotes and continue editing the SQL statement (SQL injection).
Now we I don't escape `\` it means you can escape the sql by providing
`joe\'s computer` which gets converted to `'joe\\'s computer'`
(we need something like "bind variables" to avoid this)